merkle-tree/src/db/mongodb/db.js

```diff
@@ -14,29 +14,70 @@ Class created from within src/middleware/assign-db-connection
 @param {string} contractName - contractName of the contract which relates to this db
 */
 export default class DB {
-  constructor(connection, username, contractName, treeId) {
+  constructor(connection, username, contractName, treeId, contractId) {
     this.connection = connection;
     this.username = username;
+    this.contractId = contractId;
     if (!username) return;
-    this.createModelsForUser(contractName, treeId);
+    this.createModelsForUser(contractName, treeId, contractId);
   }
 
   /**
   A model is a class with which we construct documents
   */
-  createModelsForUser(contractName, treeId) {
+  createModelsForUser(contractName, treeId, contractId) {
+    logger.debug(
+      `Creating Models. contractName: ${contractName}, treeId: ${treeId}, contractId: ${contractId}`,
+    );
+
     if (treeId === undefined || treeId === '') {
-      this.Models = {
-        node: this.connection.model(
-          `${this.username}_${contractName}_${COLLECTIONS.NODE}`,
-          nodeSchema,
-        ),
-        metadata: this.connection.model(
-          `${this.username}_${contractName}_${COLLECTIONS.METADATA}`,
-          metadataSchema,
-        ),
-      };
+      if (contractId === undefined || contractId === '') {
```

Comment:
I like this idea. While we're at it, would there be anything preventing us from getting rid of `this.username` which always resolves to `admin`? I don't see the use for it, but perhaps it'd be better to keep it for backwards compatibility?

<img width="342" alt="Screenshot 2023-07-11 at 12 02 19" src="https://github.com/EYBlockchain/timber/assets/136651054/f892c6e6-e7e8-4479-905c-7a640eaf47a8">

Reply:
I believe we can remove it as well, and I would only be concerned about backwards compatibility. Since in the zapp we use mongo as a local db, setting up as admin for the user made sense, so the `.username` isn't needed.
However I'm not sure how it would work and whether it matters in a cloud deployment?

Reply:
IMHO we should keep it for now.